### ISSUE TITLE
New slirp logic won't build under RHEL6

### ISSUE THREAD

**Issue by polaris6262** [simh/simh#242]
The new logic for slirp is not compiling properly under RedHat Enterprise Linux 6.7. Admittedly this Linux distribution uses fairly old software but RHEL6 itself is still rather common.

The commit id for the simh code I tried to build was '3be5125d'.

The OS is RHEL 6.7; gcc version is the following:

gcc version 4.4.7 20120313 (Red Hat 4.4.7-16) (GCC)

Linux kernel is the following:

Linux 2.6.32-573.7.1.el6.x86_64 #1 SMP Thu Sep 10 13:42:16 EDT 2015 x86_64 x86_64 x86_64 GNU/Linux

Errors are the following (this has been edited to remove excessively repeated lines):

gcc -march=native -m64 -std=c99 -U__STRICT_ANSI__  -O3 -finline-functions -fgcse-after-reload -fpredictive-commoning -fipa-cp-clone -fno-unsafe-loop-optimizations -fno-strict-overflow -DSIM_GIT_COMMIT_ID=3be5125d0060806a09eab78fea93f0eb9510ca75 -DSIM_COMPILER="GCC Version: 4.4.7" -I . -D_GNU_SOURCE -DUSE_READER_THREAD -DSIM_ASYNCH_IO  -DHAVE_PCREPOSIX_H -DHAVE_DLOPEN=so -DHAVE_LIBPNG -DHAVE_GLOB -DHAVE_SHM_OPEN  PDP11/pdp11_fp.c PDP11/pdp11_cpu.c PDP11/pdp11_dz.c PDP11/pdp11_cis.c PDP11/pdp11_lp.c PDP11/pdp11_rk.c PDP11/pdp11_rl.c PDP11/pdp11_rp.c PDP11/pdp11_rx.c PDP11/pdp11_stddev.c PDP11/pdp11_sys.c PDP11/pdp11_tc.c PDP11/pdp11_tm.c PDP11/pdp11_ts.c PDP11/pdp11_io.c PDP11/pdp11_rq.c PDP11/pdp11_tq.c PDP11/pdp11_pclk.c PDP11/pdp11_ry.c PDP11/pdp11_pt.c PDP11/pdp11_hk.c PDP11/pdp11_xq.c PDP11/pdp11_xu.c PDP11/pdp11_vh.c PDP11/pdp11_rh.c PDP11/pdp11_tu.c PDP11/pdp11_cpumod.c PDP11/pdp11_cr.c PDP11/pdp11_rf.c PDP11/pdp11_dl.c PDP11/pdp11_ta.c PDP11/pdp11_rc.c PDP11/pdp11_kg.c PDP11/pdp11_ke.c PDP11/pdp11_dc.c PDP11/pdp11_dmc.c PDP11/pdp11_kmc.c PDP11/pdp11_dup.c PDP11/pdp11_rs.c PDP11/pdp11_vt.c PDP11/pdp11_td.c PDP11/pdp11_io_lib.c display/display.c display/x11.c display/vt11.c scp.c sim_console.c sim_fio.c sim_timer.c sim_sock.c sim_tmxr.c sim_ether.c sim_tape.c sim_disk.c sim_serial.c sim_video.c sim_imd.c -DVM_PDP11 -I PDP11 -DHAVE_PCAP_NETWORK -I/usr/include/ -DBPF_CONST_STRING -DUSE_SHARED -DHAVE_TAP_NETWORK -Islirp -Islirp/simh -Islirp/simh/qemu -DHAVE_SLIRP_NETWORK slirp/*.c slirp/simh/*.c -DUSE_DISPLAY -I/usr/include/X11//include -lXt -lX11 -lm -Wno-deprecated-declarations -o BIN/pdp11 -lm -lrt -lpthread -lpcreposix -ldl -lpng  
In file included from slirp/simh/qemu/qapi/error.h:81,
                 from slirp/simh/qemu/option.h:35,
                 from slirp/simh/qemu/sockets.h:28,
                 from slirp/simh/config-host.h:19,
                 from slirp/slirp.h:4,
                 from slirp/arp_table.c:25:
slirp/simh/qemu/qapi-types.h:38: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:50: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:62: warning: declaration does not declare anything
.
. (many lines deleted)
.
slirp/simh/qemu/qapi-types.h:5575: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5585: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5606: warning: declaration does not declare anything
In file included from slirp/slirp.h:149,
                 from slirp/arp_table.c:25:
slirp/mbuf.h:88: warning: declaration does not declare anything
In file included from slirp/simh/qemu/qapi/error.h:81,
                 from slirp/simh/qemu/option.h:35,
                 from slirp/simh/qemu/sockets.h:28,
                 from slirp/simh/config-host.h:19,
                 from slirp/slirp.h:4,
                 from slirp/bootp.c:24:
slirp/simh/qemu/qapi-types.h:38: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:50: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:62: warning: declaration does not declare anything
.
. (many lines deleted)
.
slirp/simh/qemu/qapi-types.h:5575: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5585: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5606: warning: declaration does not declare anything
In file included from slirp/slirp.h:149,
                 from slirp/bootp.c:24:
slirp/mbuf.h:88: warning: declaration does not declare anything
In file included from slirp/simh/qemu/qapi/error.h:81,
                 from slirp/simh/qemu/option.h:35,
                 from slirp/simh/qemu/sockets.h:28,
                 from slirp/simh/config-host.h:19,
                 from slirp/slirp.h:4,
                 from slirp/cksum.c:33:
slirp/simh/qemu/qapi-types.h:38: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:50: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:62: warning: declaration does not declare anything
.
. (many lines deleted)
.
slirp/simh/qemu/qapi-types.h:5575: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5585: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5606: warning: declaration does not declare anything
In file included from slirp/slirp.h:149,
                 from slirp/cksum.c:33:
slirp/mbuf.h:88: warning: declaration does not declare anything
In file included from slirp/simh/qemu/qapi/error.h:81,
                 from slirp/simh/qemu/option.h:35,
                 from slirp/simh/qemu/sockets.h:28,
                 from slirp/simh/config-host.h:19,
                 from slirp/slirp.h:4,
                 from slirp/dnssearch.c:29:
slirp/simh/qemu/qapi-types.h:38: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:50: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:62: warning: declaration does not declare anything
.
. (many lines deleted)
.
slirp/simh/qemu/qapi-types.h:5575: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5585: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5606: warning: declaration does not declare anything
In file included from slirp/slirp.h:149,
                 from slirp/dnssearch.c:29:
slirp/mbuf.h:88: warning: declaration does not declare anything
In file included from slirp/simh/qemu/qapi/error.h:81,
                 from slirp/simh/qemu/option.h:35,
                 from slirp/simh/qemu/sockets.h:28,
                 from slirp/simh/config-host.h:19,
                 from slirp/slirp.h:4,
                 from slirp/if.c:8:
slirp/simh/qemu/qapi-types.h:38: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:50: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:62: warning: declaration does not declare anything
.
. (many lines deleted)
.
slirp/simh/qemu/qapi-types.h:5575: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5585: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5606: warning: declaration does not declare anything
In file included from slirp/slirp.h:149,
                 from slirp/if.c:8:
slirp/mbuf.h:88: warning: declaration does not declare anything
In file included from slirp/simh/qemu/qapi/error.h:81,
                 from slirp/simh/qemu/option.h:35,
                 from slirp/simh/qemu/sockets.h:28,
                 from slirp/simh/config-host.h:19,
                 from slirp/slirp.h:4,
                 from slirp/ip_icmp.c:33:
slirp/simh/qemu/qapi-types.h:38: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:50: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:62: warning: declaration does not declare anything
.
. (many lines deleted)
.
slirp/simh/qemu/qapi-types.h:5575: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5585: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5606: warning: declaration does not declare anything
In file included from slirp/slirp.h:149,
                 from slirp/ip_icmp.c:33:
slirp/mbuf.h:88: warning: declaration does not declare anything
In file included from slirp/simh/qemu/qapi/error.h:81,
                 from slirp/simh/qemu/option.h:35,
                 from slirp/simh/qemu/sockets.h:28,
                 from slirp/simh/config-host.h:19,
                 from slirp/slirp.h:4,
                 from slirp/ip_input.c:41:
slirp/simh/qemu/qapi-types.h:38: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:50: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:62: warning: declaration does not declare anything
.
. (many lines deleted)
.
slirp/simh/qemu/qapi-types.h:5575: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5585: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5606: warning: declaration does not declare anything
In file included from slirp/slirp.h:149,
                 from slirp/ip_input.c:41:
slirp/mbuf.h:88: warning: declaration does not declare anything
slirp/ip_input.c: In function ‘ip_reass’:
slirp/ip_input.c:356: error: ‘struct mbuf’ has no member named ‘m_dat’
slirp/ip_input.c:357: error: ‘struct mbuf’ has no member named ‘m_ext’
In file included from slirp/simh/qemu/qapi/error.h:81,
                 from slirp/simh/qemu/option.h:35,
                 from slirp/simh/qemu/sockets.h:28,
                 from slirp/simh/config-host.h:19,
                 from slirp/slirp.h:4,
                 from slirp/ip_output.c:41:
slirp/simh/qemu/qapi-types.h:38: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:50: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:62: warning: declaration does not declare anything
.
. (many lines deleted)
.
slirp/simh/qemu/qapi-types.h:5575: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5585: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5606: warning: declaration does not declare anything
In file included from slirp/slirp.h:149,
                 from slirp/ip_output.c:41:
slirp/mbuf.h:88: warning: declaration does not declare anything
In file included from slirp/simh/qemu/qapi/error.h:81,
                 from slirp/simh/qemu/option.h:35,
                 from slirp/simh/qemu/sockets.h:28,
                 from slirp/simh/config-host.h:19,
                 from slirp/slirp.h:4,
                 from slirp/mbuf.c:18:
slirp/simh/qemu/qapi-types.h:38: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:50: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:62: warning: declaration does not declare anything
.
. (many lines deleted)
.
slirp/simh/qemu/qapi-types.h:5575: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5585: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5606: warning: declaration does not declare anything
In file included from slirp/slirp.h:149,
                 from slirp/mbuf.c:18:
slirp/mbuf.h:88: warning: declaration does not declare anything
slirp/mbuf.c: In function ‘m_cleanup’:
slirp/mbuf.c:43: error: ‘struct mbuf’ has no member named ‘m_ext’
slirp/mbuf.c: In function ‘m_get’:
slirp/mbuf.c:73: error: ‘struct mbuf’ has no member named ‘m_dat’
slirp/mbuf.c:89: error: ‘struct mbuf’ has no member named ‘m_dat’
slirp/mbuf.c:89: error: ‘struct mbuf’ has no member named ‘m_dat’
slirp/mbuf.c:90: error: ‘struct mbuf’ has no member named ‘m_dat’
slirp/mbuf.c: In function ‘m_free’:
slirp/mbuf.c:115: error: ‘struct mbuf’ has no member named ‘m_ext’
slirp/mbuf.c: In function ‘m_cat’:
slirp/mbuf.c:141: error: ‘struct mbuf’ has no member named ‘m_ext’
slirp/mbuf.c:141: error: ‘struct mbuf’ has no member named ‘m_dat’
slirp/mbuf.c: In function ‘m_inc’:
slirp/mbuf.c:161: error: ‘struct mbuf’ has no member named ‘m_ext’
slirp/mbuf.c:162: error: ‘struct mbuf’ has no member named ‘m_ext’
slirp/mbuf.c:162: error: ‘struct mbuf’ has no member named ‘m_ext’
slirp/mbuf.c:163: error: ‘struct mbuf’ has no member named ‘m_ext’
slirp/mbuf.c:166: error: ‘struct mbuf’ has no member named ‘m_dat’
slirp/mbuf.c:168: error: ‘struct mbuf’ has no member named ‘m_dat’
slirp/mbuf.c:170: error: ‘struct mbuf’ has no member named ‘m_ext’
slirp/mbuf.c:171: error: ‘struct mbuf’ has no member named ‘m_ext’
slirp/mbuf.c: In function ‘m_copy’:
slirp/mbuf.c:204: error: ‘struct mbuf’ has no member named ‘m_ext’
slirp/mbuf.c:204: error: ‘struct mbuf’ has no member named ‘m_dat’
slirp/mbuf.c: In function ‘dtom’:
slirp/mbuf.c:230: error: ‘struct mbuf’ has no member named ‘m_ext’
slirp/mbuf.c:230: error: ‘struct mbuf’ has no member named ‘m_ext’
slirp/mbuf.c:233: error: ‘struct mbuf’ has no member named ‘m_dat’
slirp/mbuf.c:233: error: ‘struct mbuf’ has no member named ‘m_dat’
In file included from slirp/simh/qemu/qapi/error.h:81,
                 from slirp/simh/qemu/option.h:35,
                 from slirp/simh/qemu/sockets.h:28,
                 from slirp/simh/config-host.h:19,
                 from slirp/slirp.h:4,
                 from slirp/misc.c:8:
slirp/simh/qemu/qapi-types.h:38: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:50: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:62: warning: declaration does not declare anything
.
. (many lines deleted)
.
slirp/simh/qemu/qapi-types.h:5575: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5585: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5606: warning: declaration does not declare anything
In file included from slirp/slirp.h:149,
                 from slirp/misc.c:8:
slirp/mbuf.h:88: warning: declaration does not declare anything
In file included from slirp/simh/qemu/qapi/error.h:81,
                 from slirp/simh/qemu/option.h:35,
                 from slirp/simh/qemu/sockets.h:28,
                 from slirp/simh/config-host.h:19,
                 from slirp/slirp.h:4,
                 from slirp/sbuf.c:8:
slirp/simh/qemu/qapi-types.h:38: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:50: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:62: warning: declaration does not declare anything
.
. (many lines deleted)
.
slirp/simh/qemu/qapi-types.h:5575: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5585: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5606: warning: declaration does not declare anything
In file included from slirp/slirp.h:149,
                 from slirp/sbuf.c:8:
slirp/mbuf.h:88: warning: declaration does not declare anything
In file included from slirp/simh/qemu/qapi/error.h:81,
                 from slirp/simh/qemu/option.h:35,
                 from slirp/simh/qemu/sockets.h:28,
                 from slirp/simh/config-host.h:19,
                 from slirp/simh/qemu/osdep.h:28,
                 from slirp/simh/qemu/qemu-common.h:15,
                 from slirp/slirp.c:24:
slirp/simh/qemu/qapi-types.h:38: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:50: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:62: warning: declaration does not declare anything
.
. (many lines deleted)
.
slirp/simh/qemu/qapi-types.h:5575: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5585: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5606: warning: declaration does not declare anything
In file included from slirp/slirp.h:149,
                 from slirp/slirp.c:27:
slirp/mbuf.h:88: warning: declaration does not declare anything
slirp/slirp.c: In function ‘slirp_input’:
slirp/slirp.c:744: error: ‘struct mbuf’ has no member named ‘m_ext’
slirp/slirp.c:744: error: ‘struct mbuf’ has no member named ‘m_dat’
In file included from slirp/simh/qemu/qapi/error.h:81,
                 from slirp/simh/qemu/option.h:35,
                 from slirp/simh/qemu/sockets.h:28,
                 from slirp/simh/config-host.h:19,
                 from slirp/simh/qemu/osdep.h:28,
                 from slirp/simh/qemu/qemu-common.h:15,
                 from slirp/socket.c:8:
slirp/simh/qemu/qapi-types.h:38: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:50: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:62: warning: declaration does not declare anything
.
. (many lines deleted)
.
slirp/simh/qemu/qapi-types.h:5575: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5585: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5606: warning: declaration does not declare anything
In file included from slirp/slirp.h:149,
                 from slirp/socket.c:9:
slirp/mbuf.h:88: warning: declaration does not declare anything
slirp/socket.c: In function ‘sorecvfrom’:
slirp/socket.c:488: error: ‘struct mbuf’ has no member named ‘m_ext’
slirp/socket.c:488: error: ‘struct mbuf’ has no member named ‘m_dat’
slirp/socket.c:493: error: ‘struct mbuf’ has no member named ‘m_dat’
slirp/socket.c:495: error: ‘struct mbuf’ has no member named ‘m_ext’
slirp/socket.c:495: error: ‘struct mbuf’ has no member named ‘m_dat’
In file included from slirp/simh/qemu/qapi/error.h:81,
                 from slirp/simh/qemu/option.h:35,
                 from slirp/simh/qemu/sockets.h:28,
                 from slirp/simh/config-host.h:19,
                 from slirp/slirp.h:4,
                 from slirp/tcp_input.c:41:
slirp/simh/qemu/qapi-types.h:38: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:50: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:62: warning: declaration does not declare anything
.
. (many lines deleted)
.
slirp/simh/qemu/qapi-types.h:5575: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5585: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5606: warning: declaration does not declare anything
In file included from slirp/slirp.h:149,
                 from slirp/tcp_input.c:41:
slirp/mbuf.h:88: warning: declaration does not declare anything
In file included from slirp/simh/qemu/qapi/error.h:81,
                 from slirp/simh/qemu/option.h:35,
                 from slirp/simh/qemu/sockets.h:28,
                 from slirp/simh/config-host.h:19,
                 from slirp/slirp.h:4,
                 from slirp/tcp_output.c:41:
slirp/simh/qemu/qapi-types.h:38: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:50: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:62: warning: declaration does not declare anything
.
. (many lines deleted)
.
slirp/simh/qemu/qapi-types.h:5575: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5585: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5606: warning: declaration does not declare anything
In file included from slirp/slirp.h:149,
                 from slirp/tcp_output.c:41:
slirp/mbuf.h:88: warning: declaration does not declare anything
In file included from slirp/simh/qemu/qapi/error.h:81,
                 from slirp/simh/qemu/option.h:35,
                 from slirp/simh/qemu/sockets.h:28,
                 from slirp/simh/config-host.h:19,
                 from slirp/slirp.h:4,
                 from slirp/tcp_subr.c:41:
slirp/simh/qemu/qapi-types.h:38: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:50: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:62: warning: declaration does not declare anything
.
. (many lines deleted)
.
slirp/simh/qemu/qapi-types.h:5575: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5585: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5606: warning: declaration does not declare anything
In file included from slirp/slirp.h:149,
                 from slirp/tcp_subr.c:41:
slirp/mbuf.h:88: warning: declaration does not declare anything
In file included from slirp/simh/qemu/qapi/error.h:81,
                 from slirp/simh/qemu/option.h:35,
                 from slirp/simh/qemu/sockets.h:28,
                 from slirp/simh/config-host.h:19,
                 from slirp/slirp.h:4,
                 from slirp/tcp_timer.c:33:
slirp/simh/qemu/qapi-types.h:38: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:50: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:62: warning: declaration does not declare anything
.
. (many lines deleted)
.
slirp/simh/qemu/qapi-types.h:5575: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5585: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5606: warning: declaration does not declare anything
In file included from slirp/slirp.h:149,
                 from slirp/tcp_timer.c:33:
slirp/mbuf.h:88: warning: declaration does not declare anything
In file included from slirp/simh/qemu/qapi/error.h:81,
                 from slirp/simh/qemu/option.h:35,
                 from slirp/simh/qemu/sockets.h:28,
                 from slirp/simh/config-host.h:19,
                 from slirp/slirp.h:4,
                 from slirp/tftp.c:25:
slirp/simh/qemu/qapi-types.h:38: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:50: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:62: warning: declaration does not declare anything
.
. (many lines deleted)
.
slirp/simh/qemu/qapi-types.h:5575: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5585: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5606: warning: declaration does not declare anything
In file included from slirp/slirp.h:149,
                 from slirp/tftp.c:25:
slirp/mbuf.h:88: warning: declaration does not declare anything
In file included from slirp/simh/qemu/qapi/error.h:81,
                 from slirp/simh/qemu/option.h:35,
                 from slirp/simh/qemu/sockets.h:28,
                 from slirp/simh/config-host.h:19,
                 from slirp/slirp.h:4,
                 from slirp/udp.c:41:
slirp/simh/qemu/qapi-types.h:38: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:50: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:62: warning: declaration does not declare anything
.
. (many lines deleted)
.
slirp/simh/qemu/qapi-types.h:5575: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5585: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5606: warning: declaration does not declare anything
In file included from slirp/slirp.h:149,
                 from slirp/udp.c:41:
slirp/mbuf.h:88: warning: declaration does not declare anything
In file included from slirp/simh/qemu/qapi/error.h:81,
                 from slirp/simh/qemu/option.h:35,
                 from slirp/simh/qemu/sockets.h:28,
                 from slirp/simh/glib_qemu_stubs.c:42:
slirp/simh/qemu/qapi-types.h:38: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:50: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:62: warning: declaration does not declare anything
.
. (many lines deleted)
.
slirp/simh/qemu/qapi-types.h:5575: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5585: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5606: warning: declaration does not declare anything
In file included from slirp/simh/qemu/qapi/error.h:81,
                 from slirp/simh/qemu/option.h:35,
                 from slirp/simh/qemu/sockets.h:28,
                 from slirp/simh/config-host.h:19,
                 from slirp/simh/qemu/osdep.h:28,
                 from slirp/simh/qemu/qemu-common.h:15,
                 from slirp/simh/qemu/timer.h:5,
                 from slirp/simh/sim_slirp.c:38:
slirp/simh/qemu/qapi-types.h:38: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:50: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:62: warning: declaration does not declare anything
.
. (many lines deleted)
.
slirp/simh/qemu/qapi-types.h:5575: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5585: warning: declaration does not declare anything
slirp/simh/qemu/qapi-types.h:5606: warning: declaration does not declare anything
make: *** [BIN/pdp11] Error 1

**Comment by Rhialto** [comment]
I think the issue is the option "CC_STD = -std=c99" in the makefile, and anonymous unions are only officially part of C since C11.
The C compiler that comes with NetBSD up to 6.x doesn't understand -std=c11, but the one with NetBSD 7.0 does. In the mean time, -std=gnu99 might help (unless this is a problem for some other reason).

**Comment by polaris6262** [comment]
Many thanks for the tip. Changing CC_STD to '--std=gnu99' did indeed fix the problem. I figured it was a problem with C compiler since the same simh code was compiling on my many other more-recent Linux distributions.

This is not the only RHEL6 change I had to implement in the makefile; for some reason, since the upgrade from RHEL 6.6 to 6.7 a few months ago, I've had to add -lX11 to -lSDL.

**Comment by markpizz** [comment]
The gcc you're using isn't that old, I'll change the makefile's CC_STD to -std=gnu99 when building wiith gcc or clang.

**Comment by markpizz** [comment]
Actually, since I'm in the process of cleaning up the slirp changes for submission upstream (back to qemu), I've solved this with the strategy of "less is more".  simh is using the qemu slirp code essentially unmodified.  This is achieved by providing the minimal qemu API support which is actually used by the slirp code and making those pieces either do nothing or connect into simh facilities as appropriate.  This was achieved by having replacements for the the qemu include files which the slirp files reference.  The initial cut at these include files copied way more stuff from the qemu codebase than was actually used by the slirp code.  The code causing all of the above errors/warnings is in the 'way more', but unnecessary code from qemu.  I've culled the stub code and include files down to just what is actually used by the slirp code and as such removed all of the anonymous unions and structures which were being defined but never used.
